.github/workflows/build_python_3.yml

```diff
@@ -34,17 +34,54 @@ jobs:
           fetch-depth: 0
 
       - uses: actions/setup-python@v4
+        if: matrix.os != 'arm-4core-linux'
         name: Install Python
         with:
           python-version: '3.8'
 
+      - name: Install pipx
+        if: matrix.os == 'arm-4core-linux'
+        run: |
+          curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+          sudo python3 get-pip.py
+          curl -fsSL https://get.docker.com -o get-docker.sh
+          sudo sh get-docker.sh
+          sudo usermod -a -G docker $USER
+          sudo apt install -y acl python3.10-venv
+          sudo setfacl --modify user:runner:rw /var/run/docker.sock
+          python3 -m pip install pipx
+
       - name: Set up QEMU
-        if: runner.os == 'Linux'
+        if: runner.os == 'Linux' && matrix.os != 'arm-4core-linux'
         uses: docker/setup-qemu-action@v2
         with:
           platforms: all
 
-      - name: Build wheels
+      - name: Build wheels arm64
+        if: matrix.os == 'arm-4core-linux'
```

Comment:
we cannot collapse this because the run command with pipx is different?

any way we can DRY the env variables since it is basically duplicate for the one below.

Reply:
Sadly, github actions [doesn't support anchors](https://github.com/actions/runner/issues/1182)

There may be other options to inject the env variables by using other actions, or dynamically generate the yaml, but I feel like the cure will be worse than the disease.